### ISSUE TITLE
Consistency of circuit instructions and gate names

### ISSUE THREAD

**Issue by eggerdj** [Qiskit-Extensions/qiskit-cold-atom#21]
There is some inconsistency in the naming of the circuit instructions, the string name of the gate and the gate name. For example:
```python
class LYGate(SpinGate):
    r"""Rotation of the collective spin of a cold atomic Bose-Einstein
    condensate around the y-axis.

    The generating Hamiltonian of the LY gate is

    :math:`H = \omega L_y`

    where :math:`\omega` is the free gate parameter

    **Circuit symbol:**

    .. parsed-literal::

             ┌────────────┐
        q_0: ┤ RLY(omega) ├
             └────────────┘
    """

    def __init__(self, omega, label=None):
        """Create new RLY gate."""
        super().__init__("rLy", 1, [omega], label=label)

    @property
    def generator(self) -> SpinOp:
        r"""The generating Hamiltonian of the LY gate."""
        return float(self.params[0]) * SpinOp("Y")


@add_gate
def ly(self, omega, wire):
    # pylint: disable=invalid-name
    """add the RLY gate to a QuantumCircuit"""
    return self.append(LYGate(omega), [wire], [])
```
should instead by
```python
class RLYGate(SpinGate):
    r"""Rotation of the collective spin of a cold atomic Bose-Einstein
    condensate around the y-axis.

    The generating Hamiltonian of the LY gate is

    :math:`H = \omega L_y`

    where :math:`\omega` is the free gate parameter

    **Circuit symbol:**

    .. parsed-literal::

             ┌────────────┐
        q_0: ┤ RLY(omega) ├
             └────────────┘
    """

    def __init__(self, omega, label=None):
        """Create new RLY gate."""
        super().__init__("rLy", 1, [omega], label=label)

    @property
    def generator(self) -> SpinOp:
        r"""The generating Hamiltonian of the RLYGate."""
        return float(self.params[0]) * SpinOp("Y")


@add_gate
def rly(self, omega, wire):
    # pylint: disable=invalid-name
    """add the RLY gate to a QuantumCircuit"""
    return self.append(RLYGate(omega), [wire], [])
```
This also means adapting the notebooks to this notation.

**Comment by fretchen** [comment]
Suppose we follow this notation. 

I would go further:

- Also send the `rly` string to the json files as it is otherwise really weird why we have once `rLy` and once `rly`.
- Remove the r from all entanglement and multi-spin gates. So make them `lz2` etc. Otherwise the notation becomes really clumsy with weird r everywhere, that we really only need to distinguish the qubit `lx` from the qudit `lx`
- Make sure that all the json strings are the same as the ones that would be used in the circuits.

Once we agree I could provide a draft.

**Comment by LaurinFischer** [comment]
Hi Fred, 

your suggested changes are being implemented in #23. 
- The names of all gates are now completely lowercase, e.g. `rlx` etc. 
- This is also the case for the JSONs now (actually they are just built up from the gate names)
- Daniel and I would like to keep the leading "r" for all gates that can be interpreted as a rotating around a fixed axis. For example for qubits, it's common to speak of an "RZX"-type interaction which means a two-qubit rotation generated by X times Z. In analogy, the `rlzz` gate should maybe keep the "r". 

If you're happy with this we can merge #23.

**Comment by fretchen** [comment]
PR is fine IMHO.
